### ISSUE TITLE
MixinExtras is set up incorrectly

### ISSUE THREAD

**Issue by LlamaLad7** [tr7zw/Exordium#83]
You shade it without relocating it, causing <https://github.com/QuiltMC/quilt-loader/issues/315>.
You also crucially need to [mergeServiceFiles](https://imperceptiblethoughts.com/shadow/configuration/merging/#merging-service-descriptor-files) or it will break on modern forge.
You also initialize it in a `ClientModInitializer`, that's *way* too late for most stuff. You don't seem to initialize it at all on forge. You need to do it in an [IMixinConfigPlugin](https://github.com/SpongePowered/Mixin/blob/master/src/main/java/org/spongepowered/asm/mixin/extensibility/IMixinConfigPlugin.java).

**Comment by tr7zw** [comment]
Ah, wasn't made aware that it needs to be relocated. Will look into it next week(I'm off to a festival, so not at home).

**Comment by LlamaLad7** [comment]
The whole setup process is detailed in the readme.

**Comment by LlamaLad7** [comment]
You're still initializing it far too late.

**Comment by tr7zw** [comment]
You might want to rework the readme a bit. I just read 

> The final step after setting up your build script is to initialize MixinExtras. To do this, simply call ``MixinExtrasBootstrap.init();`` **somewhere suitably early** in your program.

And took this as "as early as possible"(which would be ``onInitializeClient`` in my case). But then the next sentence says that it basically *has* to be ``PreLaunch``.

**Comment by LlamaLad7** [comment]
As early as possible is preLaunch. Main entrypoint are fired after much of the game is already initialized and so after lots of mixins are already applied.

**Comment by tr7zw** [comment]
Yes, but the wording makes it seem like the Client/Common entry points are fine(and it seems to work, at least initially). Most devs probably never had to touch the preLaunch point/are even aware that it's there.

**Comment by LlamaLad7** [comment]
I think it's fine given I specifically say to use it. Regardless, you don't need to initialize it yourself anymore from the latest betas onwards.
